### PR TITLE
SAK-47087 Samigo fix exports when columns greater than 255

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java [hunk 1]
```diff
@@ -259,107 +259,67 @@ public String getDownloadFileName() {
...
-		String mimetype = "application/vnd.ms-excel;charset=UTF-8";
+		String mimetype = "application/vnd.ms-excel";
 		String extension = ".xls";
 		int columns = findColumnSize(spreadsheetData);
 		if (columns >= 255) {
 			// allows for greater than 255 columns - SAK-16560
 			mimetype = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
 			extension = ".xlsx";
-			log.info("Samigo export ("+columns+" columns): Using xlsx mimetype: " + mimetype);
+            log.info("Samigo export ({} columns): Using xlsx mimetype: {}", columns, mimetype);
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java:269]: Spacing

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java [hunk 2]
```diff
@@ -259,107 +259,67 @@ public String getDownloadFileName() {
...
-			out = response.getOutputStream();
-			getAsWorkbook(spreadsheetData).write(out);
+
+		try (OutputStream out = response.getOutputStream(); Workbook workbook = getAsWorkbook(spreadsheetData)) {
+			workbook.write(out);
 			out.flush();
-		} catch (IOException e) {
-			log.error(e.getMessage(), e);
-		} finally {
+        } catch (IOException e) {
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java:279]: Spacing

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java [hunk 3]
```diff
@@ -259,107 +259,67 @@ public String getDownloadFileName() {
...
 	    int columns = findColumnSize(spreadsheetData);
 		Workbook wb;
 		if (columns < 255) {
-            log.info("Samigo export ("+columns+" columns): Using xsl format");
+            log.info("Samigo export ({} columns): Using xls format", columns);
 		    wb = new HSSFWorkbook();
 		} else {
 		    // allows for greater than 255 columns - SAK-16560
-		    log.info("Samigo export ("+columns+" columns): Using xslx format");
+            log.info("Samigo export ({} columns): Using xlsx format", columns);
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java:299]: Spacing

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java [hunk 4]
```diff
@@ -384,39 +346,37 @@ else if (rowData.get(0).toString().equals(HEADER_MARKER)) {
...
-							} else {
-								AnswerSurveyConverter converter = new AnswerSurveyConverter();
-								String datac = converter.getAsString(null, null, data.toString());
-								// stripping html for export, SAK-17021
-								cell.setCellValue(formattedText.convertFormattedTextToPlaintext(datac));
-							}
-						}
-					}
+
+                        if (data instanceof Integer) {
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/ExportResponsesBean.java:363]: Spacing

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java [hunk 5]
```diff
@@ -86,29 +92,30 @@ public List<EnrollmentRecord> getAvailableEnrollments(String siteid, String user
...
-			Map<String, EnrollmentRecord> enrollmentMap = new HashMap();
+			Map<String, EnrollmentRecord> enrollmentMap = new HashMap<>();
 			List<CourseSection> sections = getAvailableSections(siteid, userUid);
 			for (CourseSection section : sections) {
 				List sectionEnrollments = getSectionEnrollmentsTrusted(section.getUuid(), userUid);
-				for (Iterator eIter = sectionEnrollments.iterator(); eIter.hasNext(); ) {
-					EnrollmentRecord enr = (EnrollmentRecord)eIter.next();
-					enrollmentMap.put(enr.getUser().getUserUid(), enr);
-				}
+                for (Object sectionEnrollment : sectionEnrollments) {
```
Comment (jesusmmp) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java:99]: Spacing

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java [hunk 6]
```diff
@@ -86,29 +92,30 @@ public List<EnrollmentRecord> getAvailableEnrollments(String siteid, String user
...
+                }
 			}
 			enrollments = new ArrayList<>(enrollmentMap.values());
 		}
 		return enrollments;
 	}
 
-	public List getGroupReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
+	@Override
+    public List<EnrollmentRecord> getGroupReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
```
Comment (jesusmmp) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java:110]: Spacing

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java [hunk 7]
```diff
@@ -124,14 +131,15 @@ public List getGroupReleaseEnrollments(String siteid, String userUid, String pub
 		return enrollments;
 	}
 
-	public List getAllGroupsReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
+	@Override
+    public List<EnrollmentRecord> getAllGroupsReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
```
Comment (jesusmmp) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java:135]: Spacing

**Review by ern — APPROVED** [review]

**Pull request merged**